pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -428,6 +439,13 @@ func (s *DiscoveryServer) initConnection(node *core.Node, con *XdsConnection) (f
 		return nil, err
 	}
 
+	// Based on node metadata and version, we can associate a different generator.
+	// TODO: use a map of generators, so it's easily customizable and to avoid deps
+	proxy.Active = map[string]*model.WatchedResource{}
+	if proxy.Metadata.Generator != "" {
+		proxy.XdsResourceGenerator = s.Generators[proxy.Metadata.Generator]
```

Comment:
what if generator does not exist? As far as I understand it, in this pr `s.Generators` is always empty so this would be nil and then we fall back to the old generator?

Reply:
Yes. Not sure about it - if user asks for a generator and it's not available maybe we should fail the connection.